parser/ast/misc.go

```diff
@@ -485,9 +485,16 @@ type Prepared struct {
 	StmtType      string
 	Params        []ParamMarkerExpr
 	SchemaVersion int64
-	UseCache      bool
-	CachedPlan    interface{}
-	CachedNames   interface{}
+
+	// Used for RC or ForUpdateRead Only. It represents the version used to build plan cache last time.
+	// In Rc or ForUpdateRead, we should check whether the information schema has been changed when using plan cache.
+	// If it changed, we should rebuild the plan. However, the plan is rebuilt by using the information schema from
+	// the context provider, not the latest information schema to be consistent with the non-prepare-execute sql. But we
+	// save the latest' information schema version, which is used to build cacheKey, in order to hit plan cache.
+	LastUpdatedSchemaVersion int64
```

Comment:
Do we need `LastUpdatedSchemaVersion` cached here? If the statement is not for update, we can pass 0 for the second infoshcema, and if it is for update, we can get the latest schema version from domain directly.

Reply:
Done.